src/cryptoadvance/specter/global_search.py

```diff
@@ -0,0 +1,260 @@
+import os
+import logging
+
+from flask import url_for
+
+logger = logging.getLogger(__name__)
+
+
+class HtmlElement:
+    "This is a way to reconstruct the HTML Logical UI Tree and sum up results nicely"
+
+    def __init__(
+        self,
+        parent,
+        id=None,
+        function=None,
+        children=None,
+        visible_on_endpoints="/",
+        filter_via_input_ids=None,
+    ):
+        self.parent = parent
+        if self.parent:
+            self.parent.children.add(self)
+        self.children = children if children else set()
+        self.id = id if id else set()
+        self._result = None
+        self.function = function
+        self.visible_on_endpoints = visible_on_endpoints
+        self.filter_via_input_ids = filter_via_input_ids
+
+    @property
+    def result(self):
+        if self._result:
+            return self._result
+        else:
+            return sum([child.result for child in self.children])
+
+    @result.setter
+    def result(self, result):
+        if self.children:
+            raise "Setting results is only allowed for end nodes"
```

Comment:
Raising a string? Shouldn't you raise an Exception or a descendant from an exception?